sale_coupon_limit/models/sale_coupon_rule.py

```diff
@@ -115,11 +115,12 @@ def _compute_rule_times_used(self):
                         ]
                     )
                     continue
-                salesman_limit.rule_times_used = self.env["sale.order"].search_count(
+                salesman_limit.rule_times_used = self.env["sale.order.line"].read_group(
                     [
-                        "|",
-                        ("no_code_promo_program_ids", "in", [program["id"]]),
-                        ("code_promo_program_id", "=", [program["id"]]),
-                        ("user_id", "=", salesman_limit.rule_user_id.id),
-                    ]
+                        ("coupon_program_id", "=", [program["id"]]),
+                        ("order_id.user_id", "=", salesman_limit.rule_user_id.id),
+                        ("state", "!=", "cancel"),
+                    ],
+                    ["order_id"],
+                    ["order_id"],
                 )
```

Comment:
I miss here something for counting the results (`len`?)

Reply:
Sure :)